src/gsSolver/gsMatrixOp.h

```diff
@@ -153,16 +150,16 @@ typename gsMatrixOp<Derived>::uPtr makeMatrixOp(const Eigen::EigenBase<Derived>&
   * \ingroup Solver
   */
 template <class Derived>
-typename gsMatrixOp<Derived>::uPtr makeMatrixOp(const memory::shared_ptr<Derived> & mat, bool sym=false)
+typename gsMatrixOp<Derived>::uPtr makeMatrixOp(const memory::shared_ptr<Derived> & mat)
 {
-    return memory::make_unique(new gsMatrixOp<Derived>(mat, sym));
+    return memory::make_unique(new gsMatrixOp<Derived>(mat));
 }
 
 // We need an additional guide for the compiler to be able to work well with unique ptrs
 template <class Derived>
-typename gsMatrixOp<Derived>::uPtr makeMatrixOp(memory::unique_ptr<Derived> mat, bool sym=false)
+typename gsMatrixOp<Derived>::uPtr makeMatrixOp(memory::unique_ptr<Derived> mat)
```

Comment:
This one should be possible to be removed, when the make(shared_ptr) is updated

Reply:
no, removing this does not work. here, the problem is not the move semantics but the missing ability of c++ to deduce "Derived".